### ISSUE TITLE
`consistent-variant-order` incorrect reorder suggestion for arbitrary variant + after: pseudo-element

### ISSUE THREAD

**Issue by yellowflx** [sergioazoc/oxlint-tailwindcss#12]
The `consistent-variant-order` rule incorrectly flags classes that combine arbitrary child selectors with pseudo-element variants.

**Input:**
```
[&>*[data-role="user"]]:after:right-0
```

**Suggested fix:**
```
after:[&>*[data-role="user"]]:right-0
```

The suggestion is invalid — `after:` must come *after* the arbitrary variant selector, not before it. The arbitrary variant `[&>*[data-role="user"]]` scopes which elements are targeted, and `after:right-0` applies the style to their `::after` pseudo-element. Reversing the order changes the semantics (and Tailwind may not even parse it correctly).

**Comment by sergioazoc** [comment]
Fixed in v0.6.1 ([`5449654`](https://github.com/sergioazoc/oxlint-tailwindcss/commit/5449654)).

The root cause: in Tailwind v4, variants apply left-to-right, so `after:[&>*[data-role="user"]]:right-0` generates:

```css
&::after { &>*[data-role=user] { right: 0; } }
```

Which is broken — pseudo-elements have no children. The rule was sorting `after:` (priority 60) before `[&>*[...]]` (fallback priority 71) without considering that pseudo-elements must always be the innermost variant.

**Fix**: after sorting variants by priority, pseudo-element variants (`before`, `after`, `file`, `placeholder`, `selection`, `marker`, `backdrop`, `first-line`, `first-letter`) are now always placed last (closest to the utility). This applies in both static and design system ordering modes.

This also fixes the same problem for other element-selecting variants that were affected:
- `has-[.active]:before:` — `:has()` searches descendants, pseudo-elements have none
- `aria-expanded:after:` — attribute selectors don't apply to pseudo-elements
- `data-[state=open]:before:` — same
- `open:before:` — `[open]` attribute doesn't apply to pseudo-elements
- Triple variants like `sm:[&>svg]:after:` were reordered to `sm:after:[&>svg]:`

26 new test cases added (733 total).

Thanks for reporting!
